conans/client/tools/win.py

```diff
@@ -63,24 +65,64 @@ def build_sln_command(settings, sln_path, targets=None, upgrade_project=True, bu
     return command
 
 
-def vs_installation_path(version):
-    if not hasattr(vs_installation_path, "_cached"):
-        vs_installation_path._cached = dict()
+def vs_installation_path(version, preference=None):
```

Comment:
I am thinking it twice, and I think this is an undesired interface. Recipes shouldn't code which build_tool they prefer, this is mostly environment configuration. I think I would remove the input parameter and use only config/env-vars. WDYT @lasote?